File: Python/SQLAlchemy_ORM/dal/entities/order_detail.py
from sqlalchemy import Column, Float, ForeignKey, Integer, SmallInteger, String, Text
from sqlalchemy.orm import relationship
from dal.data.base import Base

class OrderDetail(Base):
    __tablename__ = 'Order Details'

    OrderID = Column(Integer, ForeignKey('Orders.OrderID'), primary_key=True)
    ProductID = Column(Integer, ForeignKey('Products.ProductID'), primary_key=True)
    Quantity = Column(SmallInteger, default=1)
    UnitPrice = Column(Float)
    Discount = Column(Float)

    # Relationships with Order and Product
    order = relationship('Order', back_populates='order_details')
    product = relationship('Product', back_populates='order_details')

File: Go/SQLBoiler_ORM/dal/entities/orderDetail.go
package entities

type OrderDetail struct {
	OrderID   int     `db:"OrderID"`
	ProductID int     `db:"ProductID"`
	UnitPrice float64 `db:"UnitPrice"`
	Quantity  int     `db:"Quantity"`
	Discount  float64 `db:"Discount"`
}

func (OrderDetail) TableName() string {
	return "Order Details"
}
